### PR TITLE
tree-wide: introduce macros to define *_ref() and *_unref() functions

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by keszybz** [comment]
Commits look good. Let's bisect this. I'll push to #9951.
"sd-device: use structured initializers" — good on its own
I'm assuming origin/master — good, pr/9950 — bad

File: src/core/dynamic-user.c [hunk 1]
```diff
@@ -105,8 +105,8 @@ static int dynamic_user_acquire(Manager *m, const char *name, DynamicUser** ret)
         d = hashmap_get(m->dynamic_users, name);
         if (d) {
                 /* We already have a structure for the dynamic user, let's increase the ref count and reuse it */
-                d->n_ref++;
-                *ret = d;
+                if (*ret)
```
Comment (yuwata) [inline src/core/dynamic-user.c]: Ouch!

**Comment by yuwata** [comment]
@keszybz Thank you for the help. I found a mistake, which is fixed in the current version. I hope this will pass the CIs.

**Comment by yuwata** [comment]
Hmm, I noticed that the initial refcount of `dynamic_user` struct is 0, not 1... So, we cannot use `dynamic_user_ref()` in `dynamic_user_acquire()`.

**Pull request merged**